tests/publiccloud/azure_vm_cli.pm

```diff
@@ -0,0 +1,156 @@
+    $instance->ssh_assert_script_run("az group create -n $resource_group --tags '$tags'", timeout => 900);
+    record_info('Resource Group Created',$resource_group);
+
+    # Pint - command line tool to query pint.suse.com to get the current image name
+    my $image_name = script_output(qq/pint microsoft images --active --json | jq -r '[.images[] | select( .urn | contains("sles-15-sp4:gen2") )][0].urn'/);
+    die("The pint query output is empty.") unless ($image_name);
+    record_info("PINT", "Pint query: " . $image_name);
+
+    # Call Virtual Network ,Run Command and Virtual Machine Scale Set Test
+    load_cli_test($resource_group,$location,$machine_name,$sshkey,$image_name,$subid,@test_scripts);
+
+    sub load_cli_test {
+    my ($rg,$loc,$mn,$ssh,$img,$sub_id,@cli_tests) = @_;
+    #for each test it scp the test scripts and dependency scripts and execute the test
+    foreach my $test_name (@cli_tests) {
+       if ($instance->ssh_script_output(cmd => "ls $root_dir/$test_name | wc -l") == 0) {
+           record_info('Preparing permission for cli test ',$test_name);
+           assert_script_run("curl " . data_url("publiccloud/$test_name") . " -o /tmp/$test_name", 60);
+	   $instance->scp( "/tmp/$test_name", 'remote:' . "$root_dir/$test_name", 200 );
+           $instance->run_ssh_command(cmd => "chmod +x $root_dir/$test_name", timeout => 60);
+	   my $dep_test_name="N/A"
+           if ($test_name eq "azure_rbac.sh"){
+              my $dep_test_name = "rbac.config"
+           } elsif ($test_name eq "azure_arm_grp.sh") {
+              my $dep_test_name = "azure_arm_grp_template.json"
+           } elsif ($test_name eq "azure_arm_sub.sh") {
+              my $dep_test_name = "azure_arm_sub_template.json" 
+           } elsif ($test_name eq "azure_arm_mg.sh") {
+              my $dep_test_name = "azure_arm_mg_template.json"
+           }
+           if ($dep_test_name ne "N/A"){
+	       record_info('Preparing permission for cli test ',$dep_test_name);
+               assert_script_run("curl " . data_url("publiccloud/$dep_test_name") . " -o /tmp/$dep_test_name", 60);
+               $instance->scp( "/tmp/$test_name", 'remote:' . "$root_dir/$dep_test_name", 200 );
+               $instance->run_ssh_command(cmd => "chmod +x $root_dir/$dep_test_name", timeout => 60);
+           }
+       }
+       if ($test_name ne "azure_lib_fn.sh") {
+           record_info('Ready to run cli test ',$test_name);
+           my $start_cmd = $root_dir . "/$test_name $rg $loc $mn $ssh $img $sub_id";
+           my $script_output = $instance->run_ssh_command(cmd => "$start_cmd", timeout => 3000, proceed_on_failure => 1);
+       }
+    }
+    return;
+    }
+
+    sub cleanup {
+    my $cjob_id = get_current_job_id();
+    my $rs_group = "openqa-cli-test-rg-$cjob_id";
+    my $mc_name = "openqa-cli-test-vm-$cjob_id";
+
+    assert_script_run("az group delete --resource-group $resource_group --yes", 180);
+    }
+
+    sub test_flags {
```

Comment:
azure

> Please use `perlTidy`

this script will go away.. this will be integrated in azure_cli.pm. Will run perlTidy.

Reply:
Can you please keep this file in place and not mix your bash scripts with our test logic?

Reply:
As team suggested separated bash script azure_cli testing. It is currently tests are loaded in  tests/publiccloud/azure_more_cli.pm. 

Not sure. Whether this will be created as a separate job or loaded sequentially  in lib/main_publiccloud.pm right after lib/main_publiccloud.pm.